### DISCUSSION TITLE
Is there a way to get rid of JSR305 with SpotBugs annotations?

### DISCUSSION THREAD

**Discussion by alexei-osipov** [spotbugs/spotbugs#2345] [Q&A]
So far SpotBugs annotations uses JSR305 as a dependency and JSR305 uses "javax.*" package. However this package causes conflicts with some java modules (on Java 9+) and JSR305 is generally considered to be deprecated. Are there any plans or existing ways to support using SpotBugs without compile-time dependency on JSR305?

**Comment by hazendaz** [comment]
What conflicts for example are caused?  I've never seen any and have been on newer jdks a long time.

**Reply by alexei-osipov** [reply]
Using 3rd partly library with `javax` namespace causes "split package" problem if you also use any java modules that also use `javax` namespace. For example, `java.xml` build in module. See https://github.com/micronaut-projects/micronaut-core/issues/441

Also see https://nipafx.dev/jsr-305-java-9/

The "correct" approach to the problem is stopping using `javax` package for any 3rd party lib (like JSR305). So we want to get rid of JSR305 from our classpath in favor of SpotBugs annotations. However it seems that SpotBugs annotations are not "self defined" but still actually depend on JSR305 and actively use it to declare own set of annotations.

**Reply by hazendaz** [reply]
Thanks for clarification, issue is with modular.  Haven't ran into that myself yet as little to nothing modular but get it...thanks.

**Comment by guw** [comment]
FWIW, here is another example of problems caused by the existing jsr305 jar and the split-package:
https://github.com/eclipse-platform/eclipse.platform.ui/issues/727
